test_site/pages/home_with_excluded_elements.rb

```diff
@@ -0,0 +1,42 @@
+class TestHomePageWithExcludedElements < SitePrism::Page
+  set_url '/home.htm'
+  set_url_matcher(/home\.htm$/)
+
+  # individual elements
+  element :welcome_header, :xpath, '//h1'
+  element :welcome_message, :xpath, '//span'
+  element :go_button, :xpath, '//input'
+  element :link_to_search_page, :xpath, '//a'
+  element :some_slow_element, :xpath, '//a[@class="slow"]'
+  element :invisible_element, 'input.invisible'
+  element :shy_element, 'input#will_become_visible'
+  element :retiring_element, 'input#will_become_invisible'
+  element :remove_container_with_element_btn, 'input#remove_container_with_element'
+
+  # elements groups
+  elements :lots_of_links, :xpath, '//td//a'
```

Comment:
I don't generically have an issue with what you've done here to obtain Test Isolation. However I do believe that this class could be 25-50% lighter. 

To start with lets remove all plural items (`sections` and `elements`), and perhaps just have a couple of `element` and `section` objects. With of course the element objects that don't exist.

2.10 / 2.11 are going to be massively trimmed down dieted versions ideally of SitePrism so it's easier to manage going forwards.

Reply:
The more I think about it, the more I'd like to actually have a pared down htm file to match a pared-down page object class to test the `excluded_elements` case. Will add that as well.